### PR TITLE
DDF-886 - Added number support for the Query builder.

### PR DESCRIPTION
**Pull request by McLucas**
-adjusted the filter controller to handle different types numeric types
that come back from the search results.  Any java number type found will
be converted to the "number" filter type.  This saves alot of boilerplate
code and we can handle numbers in a generic way.
-removed != from the number operator list.  Its not supported on the
backend yet, so I left a placeholder for when its supported.

### CODE REVIEW

**Comment by McLucas** [comment]
JSHint errors fix.  That is what failed the travis build.

**Comment by shaundmorris** [comment]
This looks good to me.

**Comment by McLucas** [comment]
I got one more thing i'm going to add in here.  Its a fix for the contains / ILIKE partial word searches.

**Comment by McLucas** [comment]
Update PR with the ILIKE partial word search.

File: search-ui/standard/src/main/webapp/js/model/CQL.factory.js [hunk 1]
```diff
@@ -32,13 +32,19 @@ define([
...
 
-        getValue: function(value) {
+        getValue: function(value, isILIKE) {
             switch (typeof value) {
                 case 'string':
                     if(moment(value).isValid()){
                         return moment(value).format(Properties.CQL_DATE_FORMAT);
+                    } else {
+                        if(isILIKE){
+                            return "'*" + value.replace(/'/g, "''") + "*'";
```
Comment (pklinef) [inline search-ui/standard/src/main/webapp/js/model/CQL.factory.js]: I am having second thoughts about this.  If we do not add them, a user can decide to add a wild card to the front, end, both, or neither.  They lose all of those options if we hardcode it.

Comment (McLucas) [inline search-ui/standard/src/main/webapp/js/model/CQL.factory.js]: Thats a good point.  Would replacing *\* with single \* work?

Comment (pklinef) [inline search-ui/standard/src/main/webapp/js/model/CQL.factory.js]: For maximum flexibility, I would not include a pre or post wildcard in the query.  If a user needs to match a partial word, they can include the wildcard where they need it.  That will allow them to do exact matches if they need to.

Comment (McLucas) [inline search-ui/standard/src/main/webapp/js/model/CQL.factory.js]: roger.  reverted the wildcard change.

**Comment by McLucas** [comment]
This should be good to go.

**Pull request merged**